BParkingNano/plugins/KinFitter.cc

```diff
@@ -0,0 +1,143 @@
+#include "KinFitter.h"
+
+
+KinFitter::KinFitter() {};
+
+KinFitter::KinFitter( std::vector< RefCountedKinematicParticle >& allParticles,
+                      bool vrefit )
+{ 
+
+  KinematicConstrainedVertexFitter kcvFitter;    
+  bsTree = kcvFitter.fit( allParticles );
+  m_npart = allParticles.size();
+
+  // check if fit is succesfull
+  if( bsTree->isEmpty() )       { m_success=false; return; }
+  if( !bsTree->isValid() )      { m_success=false; return; }
+  if( !bsTree->isConsistent() ) { m_success=false; return; }
+  // get B
+  bsTree->movePointerToTheTop(); 
+  b_s = bsTree->currentParticle();
+  b_dec_vertex = bsTree->currentDecayVertex();
+  // more quality checks
+  if( !b_s->currentState().isValid() ) { m_success=false; return; }
+  bs_state = b_s->currentState();
+  if( !b_dec_vertex->vertexIsValid() ) { m_success=false; return; }
+  bs_children = bsTree->finalStateParticles();
+  if( bs_children.size() != m_npart )  { m_success=false; return;}
+  bs_track = b_s->refittedTransientTrack();
+  // if pass set success to true
+  m_success = true;
+  refit = vrefit;
+
+}
+
+// fit with mass constraint
+KinFitter::KinFitter( std::vector< RefCountedKinematicParticle >& allParticles,
```

Comment:
Well it was meant to be more general. As a ,matter of fact I though that in K*ll or phill it can be useful. [For my education: What "k/pi mass will be used to tah the event" means? ]

Yes I know this... The reconstruction that employs those should pass first the objects to be constraint. For Kll I am passing leptons since this will not be used. If I were to create K* mass then tracks should be the first two entries in the vector